src/programming-assistants.rst

```diff
@@ -1,23 +1,179 @@
+
 Program Repair
 ==============
 
+**Program Repair** refers to the process of automatic repairing of a code corpus that contains syntactical and/or logical errors. It is intended to speed up the time and effort exhausted to debug a software project. Debugging can consume a significant amount of time the larger or more complex the project is. Not only does the root cause of an issue have to be found but the bug itself has to be fixed making the whole process very tedious. Some common techniques used in Program Repair are statistical fault localization and component-based program synthesis.
+
+A very common approach for Program Repair is using **Genetic Programming**, a computational method inspired by biological evolution which evolves computer programs tailored to a specific task. Researchers at CMU are doing just this by combining program analysis methods with evolutionary computation to automatically repair bugs :cite:`weimer2010automatic`. The key feature about the research was that it did not rely on formal specifications, allowing it to be more flexible to a larger range of software.
+
+The work introduces algorithms to find and minimize the number of repairs required in a program based on test cases that describe the desired functionality. The algorithms are generic enough to span a broad range of bugs. The research also introduces a novel and efficient representation to apply Genetic Programming to Program Repair with experimental results showing how the approach generates repairs for several types of bugs in 11 programs. The solution was able to fix the bugs across all 11 programs (over 60,000 lines of code total) in 2000 seconds, an impressive feat considering the total size of the projects analyzed.
+
+Another Program Repair tool, SemFix is an automated program repair method based on symbolic execution, constraint solving and program synthesis. It utilizes statistical fault localization in order to identify and rank lines of code based on their suspiciousness, determines the correct specifications of buggy statements using a method similar to angelic debugging and finally uses program synthesis to correct the statement :cite:`nguyen2013semfix`.
+
+To evaluate the tool, a buggy test set of 50 was used with a total of 90 bugs. The performance and speed of SemFix was also compared to those of GenProg, a competing automatic debugging tool. SemFix proved to be more successful and faster with the average speed repair speed being 100 ms. Although SemFix outperformed GenProg, it was still only able to debug 48 of the 90 bugs successfully (GenProg was only able to debug 16 of the 90) which seems problematic. However, the SemFix team clearly addresses this and explains the drawbacks of the tool. More details can be found `here <https://www.comp.nus.edu.sg/~abhik/pdf/ICSE13-SEMFIX.pdf>`_.
+
 Interactive Proof Assistants
 ============================
+
+
+Bibliography
```

Comment:
The idea was for these items to be integrated into the text, with a short summary of each paper (like `notation.rst`). Can you do that for the second draft?